R/tbl_cross.R

```diff
@@ -0,0 +1,163 @@
+    select_single = TRUE
+  )
+
+  # matching arguments ---------------------------------------------------------
+  missing <- match.arg(missing)
+  percent <- match.arg(percent)
+
+
+  # if no x and y provided, default to first two col_strumns of data -----------
+  if (is.null(row_str) & is.null(col_str)) {
+    row_str <- names(data[, 1])
+    col_str <- names(data[, 2])
+  }
+
+  row_str <- row_str %||% names(select(data, -tidyselect::all_of(col_str)))[1]
+  col_str <- col_str %||% names(select(data, -tidyselect::all_of(row_str)))[1]
+
+  # create new dummy col for tabulating column totals in cross table
+  data <- data %>%
+    dplyr::select(tidyselect::all_of(c(row_str, col_str))) %>%
+    mutate(Total = 1)
+
+  # get col label (for tab spanner)--------------------------------------------
+  label <- tidyselect_to_list(data, label) %||% NULL
+
+  col_label <- label[[col_str]] %||%
+    attr(data[[col_str]], "label") %||% col_str
+
+  # statistic argument ---------------------------------------------------------
+
+  # if no user-defined stat, default to {n} if percent is "none"
+  statistic <- statistic %||% ifelse(percent == "none", "{n}", "{n} ({p}%)")
+  statistic_list <- rep(statistic, 3) %>%
+    as.list() %>%
+    set_names(names(data))
+
+  # omit missing data, or factorize missing level ------------------------------
+  lbls <- purrr::map(data, ~ attr(.x, "label"))
+
+  data <- data %>%
+    mutate_at(vars(row_str, col_str), ~ as.factor(.x)) %>%
+    mutate_at(
+      vars(row_str, col_str),
+      ~ switch(
+        missing,
+        "no" = .,
+        "ifany" = forcats::fct_explicit_na(., missing_text),
+        "always" = forcats::fct_explicit_na(., missing_text) %>%
+          forcats::fct_expand(missing_text)
+      )
+    )
+
+  if (missing == "no") {
+    data <- stats::na.omit(data)
+
+    message(glue(
+      "{sum(is.na(data))} observations missing data have been removed."
+    ))
+  }
+
+  # re-applying labels
+  for (i in names(lbls)) {
+    attr(data[[i]], "label") <- lbls[[i]]
+  }
+
+
+  # create main table ----------------------------------------------------------
+  x <- data %>%
+    tbl_summary(
+      by = col_str,
+      statistic = statistic_list,
+      percent = switch((percent == "none") + 1, percent, NULL),
+      label = label,
+      missing_text = missing_text
+    ) %>%
+    add_overall(last = TRUE) %>%
+    bold_labels() %>%
+    modify_header(
+      stat_by = "{level}",
+      stat_0 = " "
+    )
+
+  footnote_text <- footnote_stat_label(x$meta_data)
+
+  # gt function calls ------------------------------------------------------------
+  # quoting returns an expression to be evaluated later
+  x$gt_calls[["tab_spanner"]] <-
+    glue(
+      "gt::tab_spanner(",
+      "label = gt::md('**{col_label}**'), ",
+      "columns = contains('stat_')) %>%",
+      "gt::tab_spanner(label = gt::md('**Total**'), ",
+      "columns = vars(stat_0))"
+    )
+
+  x$gt_calls[["tab_footnote"]] <-
```

Comment:
i am not sure why this is needed?  isn't this returned from the `tbl_summary` call?

Reply:
This footnote business is so I can adjust the location of the footnote. Here I am overwriting the default tbl_summary() footnote to change the location and place it over "Characteristic" instead of the col labels because it applies to rows and columns both. There may be a more elegant way of doing this...

Reply:
Hmmm maybe with `gt::tab_source_note()`? Then there is no need to place the footnote.

Reply:
Yes, I was considering that. I'll give it a try in the next iteration.